client/finality-grandpa/src/finality_proof.rs

```diff
@@ -551,8 +765,17 @@ impl<Header: HeaderT> AuthoritiesOrEffects<Header> {
 	}
 }
 
+/// Block info extracted from the justification.
```

Comment:
After checking this trait is only useful to keep the J parameter in the check method, this is similar to existing code, but since there is no use of TestFinality it is not strictly useful. I guess the right move is to create a test case that use prove and check, but at this point I did not check if implementing it for TestFinality would be straight forward.